x/shield/keeper/purchase.go

```diff
@@ -220,11 +217,20 @@ func (k Keeper) RemoveExpiredPurchasesAndDistributeFees(ctx sdk.Context) {
 					if !found {
 						panic("cannot find the pool for an expired purchase")
 					}
+					totalShield := k.GetTotalShield(ctx)
 					totalShield = totalShield.Sub(entry.Shield)
+					k.SetTotalShield(ctx, totalShield)
 					pool.Shield = pool.Shield.Sub(entry.Shield)
 					k.SetPool(ctx, pool)
 					// Minus one because the current entry is deleted.
 					i--
+
+					if entry.ServiceFees.Native.Empty() {
+						if entry.ServiceFees.Native.Empty() && ctx.BlockHeight() > common.Update1Height {
+							k.ProcessStakeForShieldExpiration(ctx, poolPurchaser.PoolID, entry.PurchaseID, bondDenom,
```

Comment:
Why do we process it at deletion time rather than protection end time?

Reply:
Hmm wasn't sure where to put it 🤔. I think it should be protection end time. fixed